eas-cli: fix most functionalities missing and dangling symlink

zestsystem opened the pull request:
Resolves #381256 and #382259

<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [X] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

donovanglover reviewed (commented):
Please add pre/post hooks and remove the echo statements. Also one commit for each fix is ideal

eclairevoyant commented:
(and if you keep the hooks, you don't need to change the phase from `postInstall` to `postFixup`)

zestsystem committed:
eas-cli: replace yarnInstallHook to fix symlink and missing cli features

eas-cli: add preInstall and postInstall hooks

eas-cli: use postFixup instead of postInstall

chore: remove patchelf

zestsystem commented:
@donovanglover Thanks for the feedback, and I applied the changes. In this case replacing yarnInstallHook fixed both issues simultaneously, so I have it as one commit addressing both issues.

eclairevoyant commented on pkgs/by-name/ea/eas-cli/package.nix:
```diff
@@ -30,23 +28,21 @@ stdenvNoCC.mkDerivation (finalAttrs: {
-    jq '. + {version: "${finalAttrs.version}"}' package.json > package.json.tmp
-    mv package.json.tmp package.json
+  # yarnInstallHook strips out build outputs within packages/eas-cli resulting in most commands missing from eas-cli.
+  installPhase = ''
```
```suggestion
  installPhase = ''
    runHook preInstall

```

eclairevoyant commented on pkgs/by-name/ea/eas-cli/package.nix:
```diff
@@ -30,23 +28,21 @@ stdenvNoCC.mkDerivation (finalAttrs: {
+    mkdir -p $out/lib/node_modules/eas-cli-root
+    yarn install --immutable --immutable-cache --offline --non-interactive --production --frozen-lockfile
+    cp -r . $out/lib/node_modules/eas-cli-root
   '';
```
```suggestion

    runHook preInstall
  '';
```

eclairevoyant commented on pkgs/by-name/ea/eas-cli/package.nix:
```diff
@@ -30,23 +28,21 @@ stdenvNoCC.mkDerivation (finalAttrs: {
     yarnBuildHook
-    yarnInstallHook
     nodejs
+    patchelf
```
I don't see the purpose of this line, given that [`stdenv` already provides `patchelf`](https://nixos.org/manual/nixpkgs/unstable/#sec-tools-of-stdenv)
```suggestion
```

and I don't see it used

zestsystem replied:
`/nix/store/h9lc1dpi14z7is86ffhl3ld569138595-audit-tmpdir.sh: line 23: patchelf: command not found`
I get this unless I explicitly put patchelf in buildInput. I'm on nix-darwin so maybe thats why?

eclairevoyant replied:
According to <https://nixos.org/manual/nixpkgs/unstable/#audit-tmpdir.sh> this should only be run on linux. Either way adding `patchelf` on darwin doesn't work, patchelf only works on linux.

zestsystem replied:
Got it. Thank you!

eclairevoyant replied:
Also FWIW, build seems to run on x86_64-darwin without patchelf: https://github.com/NixOS/nixpkgs/pull/382261/checks?check_run_id=37496412216

Not sure what issue you encountered locally :thinking:

eclairevoyant reviewed (approved):
I'm not familiar with the app itself but packaging seems relatively fine

donovanglover commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 382261`

Logs: https://github.com/donovanglover/nixpkgs-review-gha/actions/runs/13655329306


---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>eas-cli</li>
  </ul>
</details>

---
### `aarch64-linux`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>eas-cli</li>
  </ul>
</details>

---
### `x86_64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>eas-cli</li>
  </ul>
</details>

---
### `aarch64-darwin`
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>eas-cli</li>
  </ul>
</details>
